DataDog/datadog-agent
# USM: http2: fix static table paths

### What does this PR do?



This fixes a bug in the HTTP2 decoding, where the paths hardcoded in the static table (`/` and `/index.html`) are not captured, which causes the request data to be dropped due to the lack of path.

### Motivation



### Additional Notes



### Possible Drawbacks / Trade-offs



### Describe how to test/QA your changes



Run the following client and server:

#### Client

```go
package main

import (
    "bytes"
    "crypto/tls"
    "log"
    "net"
    "net/http"

    "golang.org/x/net/http2"
)

const url = "http://localhost:8000/"

func main() {
    client := &http.Client{
        Transport: &http2.Transport{
            AllowHTTP: true,
            DialTLS: func(network, addr string, cfg *tls.Config) (net.Conn, error) {
                return net.Dial(network, addr)
            },
        },
    }

    resp, err := client.Post(url, "application/json", bytes.NewReader([]byte("test")))
    if err != nil {
        log.Fatalf("Got error: %s", err)
    }
    resp.Body.Close()
}
```

#### Server

```go
package main

import (
    "log"
    "net/http"
    "time"

    "golang.org/x/net/http2"
    "golang.org/x/net/http2/h2c"
)

func main() {
    srv := &http.Server{
        Addr: ":8000",
        Handler: h2c.NewHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
            log.Printf("Got connection: %s", r.Proto)
            w.WriteHeader(200)
            w.Write([]byte("test"))
        }), &http2.Server{}),
        IdleTimeout: 2 * time.Second,
    }

    http2.ConfigureServer(srv, nil)

    srv.ListenAndServe()
    defer srv.Close()
}
```

#### Testing

- Run the server then the client to make a request.
- Run the following: `sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/http2_monitoring | jq`

You should see the following:
```json
[
  {
    "Client": {
      "IP": "127.0.0.1",
      "Port": 59830
    },
    "Server": {
      "IP": "127.0.0.1",
      "Port": 8000
    },
    "DNS": "",
    "Path": "/",
    "Method": "POST",
    "ByStatus": {
      "200": {
        "Count": 1,
        "FirstLatencySample": 5095424,
        "LatencyP50": 0
      }
    },
    "StaticTags": 0,
    "DynamicTags": null
  }
]
```

### Reviewer's Checklist
